components/plugins/webhook_notify.rb

```diff
@@ -0,0 +1,75 @@
+=begin
+    Copyright 2010-2016 Tasos Laskos <tasos.laskos@arachni-scanner.com>
+
+    This file is part of the Arachni Framework project and is subject to
+    redistribution and commercial restrictions. Please see the Arachni Framework
+    web site for more information on licensing and terms of use.
+=end
+
+# Send a webhook notification and payload after scan.
+#
+# @author Sean Handley <sean.handley@gmail.com>
+# @version 0.1
+class Arachni::Plugins::WebhookNotify < Arachni::Plugin::Base
+
+    def run
+        wait_while_framework_running
+    end
+
+    def clean_up
+        print_status 'Sending the notification...'
+
+        request = Arachni::HTTP::Request.new url:     options[:url],
```

Comment:
Switch to `http.post()` with the same options and `mode: :sync` to block for the response.